pwem/objects/data.py

```diff
@@ -931,6 +932,39 @@ def copyItems(self, otherSet,
                 if itemDataIter is not None:
                     next(itemDataIter)  # just skip disabled data row
 
+    def createCopy(self, outputPath,
+                   prefix=None, suffix=None, ext=None,
+                   copyInfo=False, copyItems=False):
+        """ Make a copy of the current set to another location (e.g file).
+        Params:
+            outputPath: where the output file will be written.
+            prefix: prefix of the created file, if None, it will be deduced
+                from the ClassName.
+            suffix: additional suffix that will be added to the prefix with a
+                "_" in between.
+            ext: extension of the output file, be default will use the same
+                extension of this set filename.
+            copyInfo: if True the copyInfo will be called after set creation.
+            copyItems: if True the copyItems function will be called.
+        """
+        fn = prefix or self.getClassName().lower().replace('setof', '')
+
+        if suffix:
+            fn += '_%s' % suffix
+
+        fn += '.%s' % (ext or pwutils.getExt(self.getFileName()))
+
+        setPath = os.path.join(outputPath, fn)
+        pwutils.cleanPath(setPath)
+
```

Comment:
The mapper part is missing:

        import pyworkflow.mapper as pwmapper
        pwmapper.SqliteDb.closeConnection(setFn)

To be hones I do not remember why is there, but I guess it was needed.

Reply:
May be @fonsecareyna82 remembers in which sceneario we had to add the closeConnection line.

Reply:
I added this in the Mapper itself...look here https://github.com/scipion-em/scipion-pyworkflow/pull/150

Reply:
I see... thanks.

Could we keep, for now the "create" name, or have an additionally create one. Mostly in tomo plugins they are using the create() avoiving the traditional self._createSetOfXXX that was the problem for this and for the subset.

Reply:
"create" is quite ambiguous, I think that we are still on time to refactor this. I would prefer to have a more explicit name for this operation.

Reply:
By the way,  I think the original case  in  TS, we should use another name for copyItems (maybe copyTs or something else), since the copyItems change  the base signature from EMSet.

Reply:
I was aiming for a common signature, so subsets, join all set operation protocols can create, copy regardless the type.

Reply:
Yes, the createCopy have a common signature and copyItems should have it as well...I'm just pointing out, that maybe in TiltSeries, the current copyItems should have a different  name, since it does not respect  the  base signature  inherited from EMSet. 

PS:  I think  I added that myself :(